File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    if (arr.length % 2 === 1) {
       return arr.map((num,i) => i % 2 === 0? num + n: num)
    } else {
       return arr.map((num,i) => i % 2 === 1? num + n: num)
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    if len(arr) % 2 == 1:
        return [num + n if i % 2 == 0 else num for i, num in enumerate(arr)]
    else:
        return [num + n if i % 2 == 1 else num for i, num in enumerate(arr)]